### PR TITLE
[WIP] Initial version

### PR DESCRIPTION
**Pull request by elmigranto**
Missing:
- [x] actual implementation of document storing (only have in-memory for now);
- [x] `/about` and `/ping`;
- [x] usage of `cluster` like [in other services](https://github.com/j3k0/ganomede-users/blob/master/index.js);
- [ ] "dockerization".

### CODE REVIEW

**Comment by elmigranto** [comment]
This is a good start for now, I'll switch to UI for this in Admin Module tomorrow. 

@j3k0 In the meanwhile, could you look through this (maybe just [specs in README](https://github.com/j3k0/ganomede-data/blob/229a7383a211acc767e93ea04c77b49137d49182/README.md)) and let me know if anything needs changing. Things to look for:

- server-generated random ids (uuid4) which are not embedded inside documents;
- gzip is only on `GET` responses (no compatible payloads elsewhere anyways);
- create/replace bodies are `{secert: "api_secret", document: {/*this goes into db*/}}`.

**Comment by elmigranto** [comment]
Do we need any kind of searching? In general, what kind of things we are going to store? What about listing documents?

For now, we can just save a thing, remember its ID and use it later. Not super useful, given that there are no practical way to retrieve something without knowing ID. 

If we need, for example, cache some stuff here like `users:jeko:recent-games`, we will probably want an ability to generate IDs ourselves and maybe search/filter them.

If we need more complicated stuff with searching through docs themselves, we probably need to use something other than Redis, so we won't have to implement that ourselves.

For now, most we can have in UI is specifying exact ID and editing its contents, possibly with a list of documents (but those are just random uuids). Would that be enough?

**Comment by elmigranto** [comment]
In short, could you elaborate a bit on how we plan to use this module and stuff it supposed to store.

**Comment by j3k0** [comment]
The first thing we're gonna use this module for is to store dictionaries of words (for the upcoming word game we're developing).

Example of database content:

| ID | document |
| --- | --- |
| english | [ "animals", "towns" ] |
| french | [ "animaux" ] |
| english:animals | [ "dog", "cat", "fish" ] |
| english:towns | [ "london", "paris", "moscow" ] |
| english:materials | [ "rock", "sand", "glass" ] |
| french:animaux | [ "chien", "chat", "grenouille" ] |

A game server, when asked to create a game for english speakers, will load the list of public english dictionaries (ID=lang). Pick a random category, then load the words in this category (ID=lang:category).

Eventually we'll prefix the IDs with appname + appversion.

That's the use case.

> server-generated random ids (uuid4) which are not embedded inside documents;

That's cool. Except we also need to allow users to generate IDs.

> gzip is only on GET responses (no compatible payloads elsewhere anyways);

Perfect. No big payloads are expected, so it might even be slower with gzip eventually.

> create/replace bodies are {secret: "api_secret", document: {/*this goes into db*/}}.

Seems alright. I'd suggest an optional `id` field added in the create request body to solve the point talked about above.

**Comment by elmigranto** [comment]
Hey @j3k0 can you please take a look at docker-related stuff? For some reason first forked child can't start listening, but second fork succeeds. Even after that, none of the stuff actually works, all HTTP requests are empty response and ports do not seem to be forwarded correctly.

I'm giving up this battle for today :(

**Comment by j3k0** [comment]
@elmigranto : I think this was your issue. https://github.com/j3k0/ganomede-data/pull/2/commits/2c298f1e321eb13980d97585642fadf32a4db103#diff-e22b50e740e102c8de5e858dbcd7753dR24
Server bound to 127.0.0.1 inside a container, which is the host machine local interface, but the container has no access to it (except if explicitly allowed, for instance by using `--net=host` with docker). FYI, the local interface inside a container is generally something like `172.17.xxx.xxx`, but one should ignore that and bind to `0.0.0.0` by default.

File: Dockerfile [hunk 1]
```diff
@@ -20,6 +20,6 @@ RUN chown -R app /home/app
 USER app
 
 WORKDIR /home/app/code
-CMD node index.js | bunyan
+CMD node index.js
```
Comment (j3k0) [inline Dockerfile]: It's best to expose raw logs, we can run `bunyan` with custom filters on raw logs, we can't anymore once they're formatted for display.

**Pull request merged**